subsys/net/ip/l2/ieee802154/ieee802154_security.c

```diff
@@ -151,12 +158,13 @@ bool ieee802154_encrypt_auth(struct ieee802154_security_ctx *sec_ctx,
 
 	pkt.in_buf = encrypt_payload_len ? frame + auth_payload_len : NULL;
 	pkt.in_len = encrypt_payload_len;
-	pkt.out_buf = frame;
+	/* Prevent the header from being overwritten */
+	pkt.out_buf = frame + auth_payload_len;
 	pkt.out_buf_max = IEEE802154_MTU - IEEE802154_MFR_LENGTH;
 
 	apkt.ad = frame;
 	apkt.ad_len = auth_payload_len;
-	apkt.tag = NULL;
+	apkt.tag = pkt.out_buf + encrypt_payload_len; /* mbedtls needs it */
```

Comment:
get rid of the mbedtls comment. all crypto device should behave the same, so if tc/mtls or cc2520 crypto device diverge on the behaviour: they need to be fixed. (Or the crypto api needs to be fixed also)

Reply:
ignore comment, I have described it in #8339

Reply:
I'm handling #8339, and aim to handle CCM tag in the same way. Suppose this change follow API's description.